docs/development/go/go-context.md

```diff
@@ -0,0 +1,468 @@
+                err := ok.err
+                resp := ok.r
+                if err != nil {
+                        fmt.Println("Error select:", err)
+                        return err
+                }
+                defer resp.Body.Close()
+
+                realHTTPData, err := ioutil.ReadAll(resp.Body)
+                if err != nil {
+                        fmt.Println("Error select:", err)
+                        return err
+                }
+                fmt.Printf("Server Response: %s\n", realHTTPData)
+        }
+        return nil
+}
+
+func main() {
+        if len(os.Args) == 1 {
+                fmt.Println("Need a URL and a delay!")
+                return
+        }
+
+        myUrl = os.Args[1]
+        if len(os.Args) == 3 {
+                t, err := strconv.Atoi(os.Args[2])
+                if err != nil {
+                        fmt.Println(err)
+                        return
+                }
+                delay = t
+        }
+
+        fmt.Println("Delay:", delay)
+        c := context.Background()
+        c, cancel := context.WithTimeout(c, time.Duration(delay)*time.Second)
+        defer cancel()
+
+        fmt.Printf("Connecting to %s \n", myUrl)
+        w.Add(1)
+        go connect(c)
+        w.Wait()
+        fmt.Println("Exiting...")
+}
+{{< /file >}}
+
+The timeout period is defined by the `context.WithTimeout()` method in `main()`.
+The `connect()` function that is executed as a goroutine will either terminate
+normally or when the `cancel()` function is executed. 
+
+{{< note >}}
+It is considered a good practice to use `context.Background()` in the `main()` function,
+the `init()` function of a package or at tests.
+{{< /note >}}
+
+The `connect()` function is used for connecting to the desired URL. The `connect()`
+function also starts a goroutine before the `select` block takes control in order
+to either wait for web data as returned by the goroutine or for a timeout with
+the help of the `Context` variable.
+
+### Using http.go
+
+If the desired delay is too small, then `http.go` will timeout. One such example is
+when you declare that you want a delay of `0` seconds, as in the following example:
+
+        go run http.go https://www.linode.com/ 0
+{{< output >}}
+Delay: 0
+Connecting to https://www.linode.com/
+The request was cancelled!
+Exiting...
+{{< /output >}}
+
+If the timeout period is long enough, then the output from `http.go` will be similar
+to the following:
+
+        go run http.go http://localhost:8001 10
+{{< output >}}
+Delay: 1
+Connecting to http://localhost:8001
+Server Response: Serving: /
+
+Exiting...
+{{< /output >}}
+
+Notice that `http://localhost:8001` uses a custom made HTTP server that returns
+a small amount of data. However, nothing prohibits you from trying commands such
+as `go run http.go https://www.linode.com/ 10` .
+
+## Another use of Context
+
+In this section of the guide you are going to learn how to pass values in a `Context`.
```

Comment:
Good to highlight this, but with three caveats:
- The title is kind of awkward. `Another use of Context` sounds vague, when in reality this is about using contexts as key-value stores.
- The section is pretty brief compared to the others. Explaining _why_ a user might want to do this would be good.
- This section conflicts with the introductory statement that passing values into contexts is done to provide further information about why they were cancelled.

Reply:
- Changed the title
-  The example is brief so the section is brief as well
- You are right about that :)